core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -69,7 +69,7 @@ object DumpLogSegments {
 
     CommandLineUtils.checkRequiredArgs(parser, options, filesOpt)
 
-    val print = if(options.has(printOpt)) true else false
+    val print = if(options.has(printOpt) || options.has(offsetsOpt) || options.has(valueDecoderOpt) || options.has(keyDecoderOpt)) true else false
```

Comment:
Another nit: the variable `print` seems way too generic. Maybe we could rename to `printData` or something better?